packages/agent_cli/src/main.rs

```diff
@@ -1,618 +1,699 @@
-                        .arg(arg!(--type [TUNNEL_TYPE] "the tunnel type"))
-                        .arg(arg!(--name [NAME] "name of the tunnel"))
-                        .arg(arg!(<PORT_TYPE> "either \"tcp\", \"udp\", or \"both\""))
-                        .arg(arg!(<PORT_COUNT> "number of ports in a series to allocate").default_value("1"))
-                        .arg(arg!(--exact))
-                        .arg(arg!(--ignore_name))
-                )
-                .subcommand(
-                    Command::new("list")
-                        .about("List tunnels (format \"[tunnel-id] [port-type] [port-count] [public-address]\")")
-                )
-        )
-        .subcommand(
-            Command::new("run")
-                .about("Run the playit agent")
-                .arg(arg!([MAPPING_OVERRIDE] "(format \"<tunnel-id>=[<local-ip>:]<local-port> [, ..]\")").required(false).value_delimiter(','))
-        )
-        .subcommand(
-            Command::new("reset")
-                .about("removes the secret key on your system so the playit agent can be re-claimed")
-        )
-        .subcommand(
-            Command::new("secret-path")
-                .about("shows the file path where the playit secret can be found")
-        )
-        ;
-
-    #[cfg(target_os = "linux")] {
-        cmd = cmd.subcommand(Command::new("setup"));
-    }
-
-    cmd
-}
+use std::collections::{HashMap, HashSet};
+use std::error::Error;
+use std::fmt::{Display, Formatter};
+use std::net::{IpAddr, Ipv4Addr, SocketAddr, SocketAddrV4};
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::Duration;
+
+use clap::{arg, Command};
+use rand::Rng;
+use uuid::Uuid;
+
+use autorun::autorun;
+use playit_agent_core::api::api::*;
+use playit_agent_core::api::http_client::HttpClientError;
+use playit_agent_core::api::PlayitApi;
+use playit_agent_core::network::address_lookup::{AddressLookup, AddressValue};
+use playit_agent_core::tunnel::setup::SetupError;
+use playit_agent_core::tunnel_runner::TunnelRunner;
+use playit_agent_core::utils::now_milli;
+use playit_secret::PlayitSecret;
+
+use crate::match_ip::MatchIp;
+use crate::signal_handle::get_signal_handle;
+use crate::ui::{UI, UISettings};
+
+
+//imports (chroma)
+use std::process::Command as ProcessCommand;
+use std::str;
+use crossterm::style::Stylize;
+
+pub const API_BASE: &'static str = "https://api.playit.gg";
+
+pub mod util;
+pub mod autorun;
+pub mod playit_secret;
+pub mod match_ip;
+pub mod ui;
+pub mod signal_handle;
+
+
+
+
+
+//ping function !WINDOWS! (chroma)
+async fn ipcheck() -> Result<(), CliError> {
+
+    println!("Testing {}","ply.gg...".yellow());
+    let output = ProcessCommand::new("ping")
+        .arg("-n")
+        .arg("1")
+        .arg("ping.ply.gg")
+        .output();
+
+    match output {
+        Ok(output) if output.status.success() => {
+            println!("PLY.gg is {}","not blocked!".green());
+            check_ips("ip", &["na", "sa", "eu", "as", "in"]).await;

```

Comment:
let's not do domain based checking with `check_ips`. In a future PR we can add a debug command to check if DNS is functioning properly.